pkg/json/ingest.go

```diff
@@ -98,17 +98,21 @@ func (ingester *Ingester) ingest(input jsonom.Node, path ls.NodePath, schemaNode
 		}
 		return node, dp, nil
 	}
-
-	if schemaNode != nil && schemaNode.GetTypes().Has(ls.AttributeTypes.Polymorphic) {
-		return validate(ingester.ingestPolymorphicNode(input, path, schemaNode))
-	}
-	switch next := input.(type) {
-	case *jsonom.Object:
-		return validate(ingester.ingestObject(next, path, schemaNode))
-	case *jsonom.Array:
-		return validate(ingester.ingestArray(next, path, schemaNode))
+	// only ingest nodes that have a matching schema attribute
```

Comment:
Did you make sure all functions calling ingester.ingest can deal with a nil,nil,nil return value?